gdpr/purpose_config.go

```diff
@@ -0,0 +1,41 @@
+package gdpr
+
+import (
+	"github.com/prebid/go-gdpr/consentconstants"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+// purposeConfig represents all of the config info selected from the host and account configs for
+// a particular purpose needed to determine legal basis using one of the GDPR enforcement algorithms
+type purposeConfig struct {
+	PurposeID                  consentconstants.Purpose
+	EnforceAlgo                string
```

Comment:
we seem to do a lot of comparisons with the string, and one instance of printing it out. It should only take on 2 values (3 if you count null/empty). Would it be better to set up some constants with iota, and then have a string lookup map or array to get the human readable string the one time we need it, rather than having multiple string comparisons executed for each request?

Reply:
Agreed. I added an enforcement algorithm integer representation using iota to the host config and account config. The host config is populated on startup and the account config value is set each time a request is fetched. This significantly reduces the number of string compares per request down to 10 (one per purpose) when an account is fetched. We can get the number down to 0 in the future when we refactor fetchers to cache structs instead of json.